### ISSUE TITLE
Otter-generated notebooks fail validation in Jupyter Lab/Notebook

### ISSUE THREAD

**Issue by richardcherron** [ucbds-infra/otter-grader#268]
Notebooks generated with `otter assign` fail notebook validation when opened in either Jupyter Lab or Jupyter Notebook. I receive the following warning in my server terminal:

```
[E 2021-07-17 11:33:02.135 ServerApp] Notebook JSON is invalid: Additional properties are not allowed ('id' was unexpected)

    Failed validating 'additionalProperties' in code_cell:

    On instance['cells'][0]:
    {'cell_type': 'code',
     'execution_count': None,
     'id': 'd5a9ad42',
     'metadata': {'deletable': False, 'editable': False},
     'outputs': ['...0 outputs...'],
     'source': '# Initialize Otter\n'
               'import otter\n'
               'grader = otter.Notebook("quiz-02...'}
```

Some students receive this warning as a browser popup in Jupyter Lab. As far as I can tell, this warning does not affect public-test grading locally or on public- and private-text grading on Gradescope.com. As far as I can tell, re-saving a notebook prevents future warnings.

**To Reproduce**

I attached the notebook and data file necessary to run `otter assign quiz-02.ipynb dist` and generate the notebooks that give this warning.

**Expected behavior**

I expect no warnings.

**Versions**

Python 3.8.10
IPython 7.22.0
Otter 2.2.4

I run `otter assign` on Win 10, but students receive this warning on both Mac and Win.

[quiz-02-for-github.zip](https://github.com/ucbds-infra/otter-grader/files/6834953/quiz-02-for-github.zip)

**Comment by hno2** [comment]
I can confirm this issue on Mac (both for generation and checking in Jupyter Notebook):
`Notebook validation failed: Additional properties are not allowed ('id' was unexpected):`

**Versions:**
* Python 3.9.4
* Otter 2.2.4
* Jupyter Notebook 6.4.0

**Comment by chrispyles** [comment]
This is a known issue. Otter uses nbformat to read in the notebook, which updates the schema of the notebook to match the latest version of the Jupyter notebook format. However, the notebook validation tools used by the classic notebook interface have not yet been updated to reflect the new notebook schema.

To fix this, pin nbformat < 5.1.0 in your environment and the error should go away.

**Comment by chrispyles** [comment]
Actually, upon further investigation, it looks like the notebook interface may have updated to allow the cell `id` attribute.

Possibly germane: https://github.com/jupyter/notebook/issues/6001

Not sure what the issue is here, but it's some issue with nbformat or juptyer notebook. I would open an issue the the repo linked above.
